subprojects/base-services/src/main/java/org/gradle/internal/classloader/FilteringClassLoader.java

```diff
@@ -46,8 +48,17 @@
 
     static {
         EXT_CLASS_LOADER = ClassLoaderUtils.getPlatformClassLoader();
-        JavaMethod<ClassLoader, Package[]> method = JavaReflectionUtil.method(ClassLoader.class, Package[].class, "getPackages");
-        Package[] systemPackages = method.invoke(EXT_CLASS_LOADER);
+        Package[] systemPackages;
+        try {
+            Method method = ClassLoader.class.getMethod("getDefinedPackages");
+            systemPackages = (Package[]) method.invoke(EXT_CLASS_LOADER);
+        } catch (NoSuchMethodException e) {
+            // Ignore
```

Comment:
This is a minor comment, but instead of "ignore", I would just explain _why_ we ignore. That is to say:

```
// most likely not running JDK 9+, trying to locate method `getPackages` instead
```